receiver/datadogreceiver/translator.go

```diff
@@ -50,60 +49,75 @@ func toTraces(payload *pb.TracerPayload, req *http.Request) ptrace.Traces {
 	for _, p := range payload.GetChunks() {
 		traces = append(traces, p.GetSpans())
 	}
-	dest := ptrace.NewTraces()
-	resSpans := dest.ResourceSpans().AppendEmpty()
-	resource := resSpans.Resource()
-	addResourceData(req, &resource)
-	resSpans.SetSchemaUrl(semconv.SchemaURL)
+	sharedAttributes := pcommon.NewMap()
+	for k, v := range map[string]string{
+		semconv.AttributeContainerID:           payload.ContainerID,
+		semconv.AttributeTelemetrySDKLanguage:  payload.LanguageName,
+		semconv.AttributeProcessRuntimeVersion: payload.LanguageVersion,
+		semconv.AttributeDeploymentEnvironment: payload.Env,
+		semconv.AttributeHostName:              payload.Hostname,
+		semconv.AttributeServiceVersion:        payload.AppVersion,
+		semconv.AttributeTelemetrySDKName:      "Datadog",
+		semconv.AttributeTelemetrySDKVersion:   payload.TracerVersion,
+	} {
+		if v != "" {
+			sharedAttributes.PutStr(k, v)
+		}
+	}
+
+	for k, v := range payload.Tags {
+		if k = translateDataDogKeyToOtel(k); v != "" {
+			sharedAttributes.PutStr(k, v)
+		}
+	}
+
+	upsertHeadersAttributes(req, sharedAttributes)
+
+	// Creating a map of service spans to slices
+	// since the expectation is that `service.name`
+	// is added as a resource attribute in most systems
+	// now instead of being a span level attribute.
+	groupByService := make(map[string]ptrace.SpanSlice)
 
 	for _, trace := range traces {
-		ils := resSpans.ScopeSpans().AppendEmpty()
-		ils.Scope().SetName("Datadog-" + req.Header.Get("Datadog-Meta-Lang"))
-		ils.Scope().SetVersion(req.Header.Get("Datadog-Meta-Tracer-Version"))
-		spans := ptrace.NewSpanSlice()
-		spans.EnsureCapacity(len(trace))
 		for _, span := range trace {
-			newSpan := spans.AppendEmpty()
+			slice, exist := groupByService[span.Service]
```

Comment:
did you compare performance between main and your branch?

Reply:
I actually didn't, my bad. Let me quickly run them.

So the best branch results I got locally are:

```
pkg: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/datadogreceiver
BenchmarkTranslatorv05-10    	  213355	      5615 ns/op
BenchmarkTranslatorv07-10    	  210510	      5653 ns/op
```

Verses what is on main.
```
pkg: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/datadogreceiver
BenchmarkTranslatorv05-10    	  215205	      5133 ns/op
BenchmarkTranslatorv07-10    	  201384	      5791 ns/op
```

Extra 500ns for the v5 but a speed reduction in v7 (ever so slightly). 

Happy to spend some time later looking to improve the speed of these but for the time being, it looks like an additional 500ns isn't the most drastic change.

Reply:
Yeah for sure, just wanted to be aware of it.  A lot small changes add up to a big one, this is no big deal though.  Thanks!